src/Prometheus/Storage/APC.php

```diff
@@ -91,11 +91,21 @@ public function updateGauge(array $data): void
      */
     public function updateCounter(array $data): void
     {
-        $new = apcu_add($this->valueKey($data), 0);
-        if ($new) {
+        $valueKey = $this->valueKey($data);
+        // Check if value key already exists
+        if (apcu_exists($this->valueKey($data)) === false) {
+            apcu_add($this->valueKey($data), 0);
             apcu_store($this->metaKey($data), json_encode($this->metaData($data)));
         }
-        apcu_inc($this->valueKey($data), $data['value']);
+
+        // Taken from https://github.com/prometheus/client_golang/blob/66058aac3a83021948e5fb12f1f408ff556b9037/prometheus/value.go#L91
+        $done = false;
+        while (!$done) {
+            $old = apcu_fetch($valueKey);
+            if ($old !== false) {
+                $done = apcu_cas($valueKey, $old, $this->toInteger($this->fromInteger($old) + $data['value']));
```

Comment:
am I correct in assuming the `toInteger` method is to avoid storing a float, and we then convert it back to float?
Might be worth noting that in the commit, or maybe renaming the method to be more expicit.

Reply:
Yep, apcu can only store integers. I was "confused" by this too a bit at the first view, but then I understood it. It packs the float value as its binary represent which is then an int. The methods was used before and i just reused it but yeah it makes sense to clarify it.

Reply:
@rdohms i renamed them, could you please have a look again?